bindings/java/native/src/bee_types/payloads.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_wallet::message::MessagePayload as MessagePayloadRust;
+
+use crate::bee_types::{index::*, milestone::*, receipt::*, transaction::*, treasury::*};
+
+pub enum MessagePayloadType {
+    Transaction = 1,
+    Milestone = 2,
+    Indexation = 3,
+    Receipt = 4,
+    TreasuryTransaction = 5,
+}
+
+pub struct MessagePayload {
+    payload: MessagePayloadRust,
+}
+
+impl MessagePayload {
+    pub fn get_internal(self) -> MessagePayloadRust {
+        self.payload
+    }
+
+    pub fn new_with_internal(payload: MessagePayloadRust) -> Self {
```

Comment:
instead of `new_with_internal`, we should implement the From trait:

```rust
impl From<MessagePayloadRust> for MessagePayload {
  fn from(payload: MessagePayloadRust) -> Self {
  	Self { payload }
  }
}

// usage:
let rust_payload = MessagePayloadRust { ... };
let payload = rust_payload.into(); // or MessagePayload::from(rust_payload)
```

Reply:
this is valid for all `new_with_internal` and `new_with_rust` functions